### ISSUE TITLE
Three.js for Native Toolchain and Other Build Targets

### ISSUE THREAD

**Issue by jkassis** [soypat/three#5]
I'm writing an experimental Game Animation System and came across your really neat project. Thank you for this and all your blogs. It would be great to have one 3D interface to code against for native builds and web builds.

* Can use this with the native toolchain to build WASM?
* Can you estimate the effort to port this so that it works for regular build targets?

See also my project at...
https://github.com/jkassis/gas

**Comment by soypat** [comment]
Hey jkassis! 

Yes, you can build to wasm. You may try this out by cloning this project, installing wasmserve and running it inside one of the example directories and heading over to http://localhost:8080. The [instructions are in the examples directory](https://github.com/soypat/three/blob/main/examples/README.md).

As for your second question- I'm not sure I understand? Porting it? This is meant to run in the browser. Sure you can build the wasm code on any platform but the browser would be required to leverage three.js.

BTW I'm not the one who wrote the blogs! That may be the original author of the gopherjs version of this library...

**Comment by jkassis** [comment]
Hi soypat! Thank you. And I understand. The second question is more along the lines of... what's the effort to rewrite three.js in GoLang so that the GoLang wasm target calls WebGL through the wasm interface rather than three.js through the wasm interface.

**Comment by soypat** [comment]
I'd believe the effort is quite large. You'd have to reimplement three.js from scratch more or less.

On a related note: I've been told by the lead maintainer of the Mach Engine, Stephen Gutenkanst, that he envisions a future where Zig could serve as the interoperable interface for WebGL in Go. Stephen is the kind of person that knows his stuff so maybe I'd look into that. If you are interested in reaching him you could try the Zig discord server or via github as *slimsag*.

**Comment by gedw99** [comment]
Thread js in pure golang would cool. There is probably a generator

@jkassis 

btw I use this 
https://github.com/rajveermalviya/go-webgpu Instead of sdl . 

It targets webgpu

**Comment by jkassis** [comment]
I agree. Three.js in golang would be very nice. For 3D in golang, I've seen
https://github.com/g3n/engine, but haven't tried it yet.
